sdks/java/io/tika/src/main/java/org/apache/beam/sdk/io/tika/TikaSource.java

```diff
@@ -0,0 +1,466 @@
+    checkArgument(!metadata.isEmpty(),
+      "Unable to find any files matching %s", fileOrPattern);
+
+    return metadata;
+  }
+
+  /**
+   *  FilePatternTikaReader.
+   *  TODO: This is mostly a copy of FileBasedSource internal file-pattern reader
+   *        so that code would need to be generalized as part of the future contribution
+   */
+  static class FilePatternTikaReader extends BoundedReader<String> {
+    private final TikaSource source;
+    final ListIterator<TikaReader> fileReadersIterator;
+    TikaReader currentReader = null;
+
+    public FilePatternTikaReader(TikaSource source, List<TikaReader> fileReaders) {
+      this.source = source;
+      this.fileReadersIterator = fileReaders.listIterator();
+    }
+
+    @Override
+    public boolean start() throws IOException {
+      return startNextNonemptyReader();
+    }
+
+    @Override
+    public boolean advance() throws IOException {
+      checkState(currentReader != null, "Call start() before advance()");
+      if (currentReader.advance()) {
+        return true;
+      }
+      return startNextNonemptyReader();
+    }
+
+    private boolean startNextNonemptyReader() throws IOException {
+      while (fileReadersIterator.hasNext()) {
+        currentReader = fileReadersIterator.next();
+        if (currentReader.start()) {
+          return true;
+        }
+        currentReader.close();
+      }
+      return false;
+    }
+
+    @Override
+    public String getCurrent() throws NoSuchElementException {
+      return currentReader.getCurrent();
+    }
+
+    @Override
+    public Instant getCurrentTimestamp() throws NoSuchElementException {
+      return currentReader.getCurrentTimestamp();
+    }
+
+    @Override
+    public void close() throws IOException {
+      if (currentReader != null) {
+        currentReader.close();
+      }
+      while (fileReadersIterator.hasNext()) {
+        fileReadersIterator.next().close();
+      }
+    }
+
+    @Override
+    public TikaSource getCurrentSource() {
+      return source;
+    }
+  }
+
+  static class TikaReader extends BoundedReader<String> {
+    private ExecutorService execService;
+    private final ContentHandlerImpl tikaHandler = new ContentHandlerImpl();
+    private String current;
+    private TikaSource source;
+    private String filePath;
+    private TikaIO.Read spec;
+    private org.apache.tika.metadata.Metadata tikaMetadata;
+    private Iterator<String> metadataIterator;
+
+    TikaReader(TikaSource source, String filePath) {
+      this.source = source;
+      this.filePath = filePath;
+      this.spec = source.getTikaInputRead();
+    }
+
+    @Override
+    public boolean start() throws IOException {
+      final InputStream is = TikaInputStream.get(Paths.get(filePath));
+      TikaConfig tikaConfig = null;
+      if (spec.getTikaConfigPath() != null) {
+        try {
+          tikaConfig = new TikaConfig(spec.getTikaConfigPath().get());
```

Comment:
Is TikaConfig able to read non-local files? It seems like this will not work anywhere except in the direct runner.